packages/docusaurus/src/server/routes.ts

```diff
@@ -123,23 +123,9 @@ ${indent(subroutesCodeStrings.join(',\n'))}
   }
 
   Object.entries(props).forEach(([propName, propValue]) => {
-    // Inspired by https://github.com/armanozak/should-quote/blob/main/packages/should-quote/src/lib/should-quote.ts
-    const shouldQuote = ((key: string) => {
-      // Pre-sanitation to prevent injection
-      if (/[.,;:}/\s]/.test(key)) {
-        return true;
-      }
-      try {
-        // If this key can be used in an expression like ({a:0}).a
-        // eslint-disable-next-line no-eval
-        eval(`({${key}:0}).${key}`);
-        return false;
-      } catch {
-        return true;
-      }
-    })(propName);
-    // Escape quotes as well
-    const key = shouldQuote ? JSON.stringify(propName) : propName;
+    const isIdentifier =
```

Comment:
what about the test 😅 not really confident it works here, but it looks like it does

Reply:
We already have decent tests for this, plus this is a direct translation of the spec, so I won't worry about it too much.